Jenkinsfile

```diff
@@ -17,54 +17,37 @@
 @Library('github.com/fabric8io/fabric8-pipeline-library@master')
 def utils = new io.fabric8.Utils()
 def flow = new io.fabric8.Fabric8Commands()
-dockerTemplate{
-   clientsNode {
-    ws{
-      checkout scm
 
-      if (utils.isCI()){
-        def snapshotImageName = "fabric8/maven-builder:SNAPSHOT-${env.BRANCH_NAME}-${env.BUILD_NUMBER}"
-        container('docker'){
-          stage('build snapshot image'){
-            sh "docker build -t ${snapshotImageName} ."
-          }
-          stage('push snapshot image'){
-            sh "docker push ${snapshotImageName}"
-          }
-        }
-        stage('notify'){
-            def pr = env.CHANGE_ID
-            if (!pr){
-                error "no pull request number found so cannot comment on PR"
-            }
-            def message = "snapshot maven-builder image is available for testing.  `docker pull ${snapshotImageName}`"
-            container('docker'){
-                flow.addCommentToPullRequest(message, pr, 'fabric8io-images/maven-builder')
-            }
-        }
-      } else if (utils.isCD()){
-        
-        def v = getNewVersion{}
-        // stage('tag'){
-        //   container('clients'){
-        //     gitTag{
-        //       releaseVersion = v
-        //       skipVersionPrefix = true
-        //     }
-        //   }
-        // }
+dockerTemplate {
+    clientsNode {
+        ws {
+            checkout scm
+
+            def version = utils.isCI() ? "SNAPSHOT-${env.BRANCH_NAME}-${env.BUILD_NUMBER}" : getNewVersion{}
+            def image = "fabric8/maven-builder:${version}"
 
-        def imageName = "fabric8/maven-builder:${v}"
+            echo "Building image ${image}"
 
-        container('docker'){
-          stage('build image'){
-            sh "docker build -t ${imageName} ."
-          }
-          stage('push image'){
-            sh "docker push ${imageName}"
-          }
+            container('docker') {
+
+                stage('Build docker image') {
+                    sh "docker build -t ${image} ."
```

Comment:
did you consider using `--rm` or `--force-rm` option? I am not sure how the images are garbage collected or does it not matter at all since the `container('docker')` is ephemeral ?